### PR TITLE
Initialize the PWD variable only in sim makefile

### LINKED ISSUE #1299: Some sim Makefiles relies on automatically defined $(PWD)
## Context
Having a complex rtl module/entity requires the implementation of multiple unit-test suites, i.e. multiple standalone CoCoTb testbenches.
Each of them leaves in its own folder with its own `Makefile`.

To run all of them from a _master_ Makefile I created a file structure like the one in the attachment.
The master Makefile would then find all folder containing a CoCoTb test-bench and locally run each of their make files.

In the current implmentation this approach would fail (using **ModelSim**) as the software is not able to find the target python test-bench.

## The Issue
Inspecting the command line output I noticed that the problem relies in the definition of the `PYTHONPATH`, which contains among all, the current-working-directory.

It seems that `make` initializes the PWD variable with the path where the first Makefile it executes is located.
Hence in the particular case mentioned above, the Makefile leaving in the dedicated unit-test-suite folder will use the PWD of the master Makefile instead of its own.

In some simulator Makefiles the `PWD` is initialized with `$(shell pwd)`, in others (like the **ModelSim** one) it is not initialized at all.

## Proposed Solution
Adding the line `PWD := $(shell pwd)` in each simulator Makefile would solve this issue.

## Extra info
I use:

- Python 3.7.4
- ModelSim 10.5b  - Intel FPGA Starter Edition
- cocotb commit d06f817c669de4f02a08dfbbac35445c5b9aaeee

[make_local_pwd.zip](https://github.com/cocotb/cocotb/files/4066031/make_local_pwd.zip)

### PR DESCRIPTION
**Pull request by sfavazza-duagon**
Explicitly initialize the PWD variable in all simulation Makefiles.

Fixes #1299

### CODE REVIEW

**Comment by themperek** [comment]
Thank you for your PR.

It looks like the right thing to do but would it not better to add this to `cocotb/share/makefiles/Makefile.sim` once and remove from `Makefile.icarus` from consistency?

**Comment by sfavazza-duagon** [comment]
I think so, I will push it soon.

**Comment by themperek** [comment]
It is ok but if you could remove empty lines in individual simulator Makefiles (to have a cleaner history) and squash the commits.

File: cocotb/share/makefiles/simulators/Makefile.aldec [hunk 1]
```diff
@@ -170,6 +170,7 @@ LIB_LOAD := LD_LIBRARY_PATH=$(LIB_DIR):$(LD_LIBRARY_PATH)
 NEW_PYTHONPATH := $(PYTHONPATH)
 endif
 
+
```
Comment (eric-wieser) [inline cocotb/share/makefiles/simulators/Makefile.aldec]:
```suggestion
```

File: cocotb/share/makefiles/simulators/Makefile.cvc [hunk 2]
```diff
@@ -57,6 +57,7 @@ ifeq ($(CVC_ITERP),1)
     CVC_ARGS    += +interp
 endif
 
+
```
Comment (eric-wieser) [inline cocotb/share/makefiles/simulators/Makefile.cvc]:
```suggestion
```

File: cocotb/share/makefiles/simulators/Makefile.ghdl [hunk 3]
```diff
@@ -56,6 +56,7 @@ GHDL_ARGS += $(EXTRA_ARGS)
 
 .PHONY: analyse
 
+
```
Comment (eric-wieser) [inline cocotb/share/makefiles/simulators/Makefile.ghdl]:
```suggestion
```

File: cocotb/share/makefiles/simulators/Makefile.ius [hunk 4]
```diff
@@ -97,6 +97,7 @@ endif
 
 GPI_LIB = $(COCOTB_VPI_LIB) $(COCOTB_VHPI_LIB)
 
+
```
Comment (eric-wieser) [inline cocotb/share/makefiles/simulators/Makefile.ius]:
```suggestion
```

File: cocotb/share/makefiles/simulators/Makefile.nvc [hunk 5]
```diff
@@ -36,6 +36,7 @@ RTL_LIBRARY ?= work
 analyse: $(VHDL_SOURCES) $(SIM_BUILD)
 	cd $(SIM_BUILD) && $(CMD) --work=$(RTL_LIBRARY) -a $(VHDL_SOURCES) $(COMPILE_ARGS) $(EXTRA_ARGS)
 
+
```
Comment (eric-wieser) [inline cocotb/share/makefiles/simulators/Makefile.nvc]:
```suggestion
```

File: cocotb/share/makefiles/simulators/Makefile.questa [hunk 6]
```diff
@@ -141,6 +141,7 @@ ifneq ($(ARCH),i686)
 CMD += -64
 endif
 
+
```
Comment (eric-wieser) [inline cocotb/share/makefiles/simulators/Makefile.questa]:
```suggestion
```

File: cocotb/share/makefiles/simulators/Makefile.vcs [hunk 7]
```diff
@@ -75,6 +75,7 @@ $(SIM_BUILD)/simv: $(SIM_BUILD) $(VERILOG_SOURCES) $(SIM_BUILD)/pli.tab $(COCOTB
 	-timescale=$(COCOTB_HDL_TIMEUNIT)/$(COCOTB_HDL_TIMEPRECISION) \
 	$(EXTRA_ARGS) -debug -load libvpi.so $(COMPILE_ARGS) $(VERILOG_SOURCES)
 
+
```
Comment (eric-wieser) [inline cocotb/share/makefiles/simulators/Makefile.vcs]:
```suggestion
```

File: cocotb/share/makefiles/simulators/Makefile.xcelium [hunk 8]
```diff
@@ -104,6 +104,7 @@ endif
 
 GPI_LIB = $(COCOTB_VPI_LIB) $(COCOTB_VHPI_LIB)
 
+
```
Comment (eric-wieser) [inline cocotb/share/makefiles/simulators/Makefile.xcelium]:
```suggestion
```

**Comment by sfavazza-duagon** [comment]
I pushed the requested change.
I have locally merged the master branch into mine using the --squash flag.
Is there anything else required on my side?
I don't know how to edit this merge request so that it automatically squashes the commits.

**Comment by themperek** [comment]
There is something wrong see: https://github.com/cocotb/cocotb/pull/1300/files

**Comment by sfavazza-duagon** [comment]
What exactly?
I just merged the master branch with mine.
Of all the changes you cited, I applied only the PWD ones.
The issue seems related to the documentation generation.

**Comment by eric-wieser** [comment]
GitHub seems to be at fault here

**Comment by eric-wieser** [comment]
I'll see if I can trick it into playing ball later today

**Comment by sfavazza-duagon** [comment]
Any news on this issue?
I don't know whether it is an issue about the tools to build the doc, but in case was a temporary server issue, is there a way to re-run the checks without pushing any commit?

**Comment by eric-wieser** [comment]
Yes, by doing that

**Comment by eric-wieser** [comment]
Hmm, still a mess

Can you rebase locally, and force push?

**Comment by sfavazza-duagon** [comment]
Ok I think I got it right now (at least the git part 😅 )

**Comment by eric-wieser** [comment]
I'm now a bit confused by the title of this PR - is it out of date?

**Comment by sfavazza-duagon** [comment]
Can you clarify?
I re-based exactly the same commits as when I opened this issue.

**Comment by eric-wieser** [comment]
PR title is:

> Calculate the PWD variable locally (do not use the Makefile one)

Yet the change seems to be removing the local calculation, and changing it so it _does_ use the (top-level) makefile one

**Comment by sfavazza-duagon** [comment]
The issue started like that, then drifted toward the better solution of calculate the PWD variable once and for all.
Feel free to change it to what you think it's most suitable for you.

**Comment by eric-wieser** [comment]
So, context for why I ask that - normally we'll squash commits with a review history like this, but when I squash commits, I need a new commit message that's consistent with the [Patch Requirements](https://github.com/cocotb/cocotb/blob/master/CONTRIBUTING.md#patch-requirements).

I'm generally happy to do that for you instead of making you update your commit messages, but it's a lot easier for me to do that if I can just copy the PR title and description :). It also makes our changelog easier to follow. If you edit those, then I'll put them in the final commit message.

**Comment by sfavazza-duagon** [comment]
Ok I see, check if the new one is good (I thought I perfectly digested the contribution guidelines, doh 😓).

**Comment by eric-wieser** [comment]
Should this line be removed too?

https://github.com/cocotb/cocotb/blob/65633e2c989e8e6e4204fe9171d40b67b5bd6c23/cocotb/share/makefiles/simulators/Makefile.verilator#L51

**Comment by sfavazza-duagon** [comment]
I removed it from all simulator makefiles as the PWD will be calculated only in `Makefile.sim` which in turn will call `Makefile.verilator`.
So yes, I think it is correct.

**Comment by eric-wieser** [comment]
Can you push an  update with that line removed then?

**Comment by sfavazza-duagon** [comment]
Any other objections?

**Pull request merged**